docs/parameters.md

```diff
@@ -0,0 +1,324 @@
+# Passing parameters to brokers
+
+Table of Contents
+- [Overview](#overview)
+- [Design](#design)
+  - [Basic example](#basic-example)
+  - [Passing sensitive data](#passing-sensitive-data)
+  - [Value types](#value-types)
+    - [String](#string)
+    - [JSON object](#json-object)
+    - [Other primitive types (int, bool, float)](#other-primitive-types-int-bool-float)
+    - [Arrays](#arrays)
+  - [Passing all parameters from a single source at once](#passing-all-parameters-from-a-single-source-at-once)
+  - [Conflict resolution](#conflict-resolution)
+
+## Overview
```

Comment:
I think that adding `type: raw` should be sufficient.
I don't want to extend the union types, as I think they would be too confusing.